### PR TITLE
feat(gatsby-source-contentful): add structured text node type

### PR DESCRIPTION
**Pull request by Khaledgarbaya**
This PR adds the support for a new field in contentful called [rich text](https://www.contentful.com/developers/docs/concepts/rich-text/)

> The Rich Text is a new field type that enables a rich text editor in an entry. It allows a content author to format text in a visual way (WYSIWYG) and also embed linked entries as well. 

This is still in a alpha and super early. This Pr is just adding the field type to the schema so we can create transformers for it in the future.

### CODE REVIEW

**Comment by pieh** [comment]
@Khaledgarbaya let us know if we should merge it as-is or keep this open

File: packages/gatsby-source-contentful/src/normalize.js [hunk 1]
```diff
@@ -373,6 +393,21 @@ exports.createContentTypeNodes = ({
           childrenNodes.push(textNode)
           entryItemFields[`${entryItemFieldKey}___NODE`] = textNode.id
 
+          delete entryItemFields[entryItemFieldKey]
+        } else if (
+          fieldType === `StructuredText` &&
+          _.isPlainObject(entryItemFields[entryItemFieldKey])
```
Comment (KyleAMathews) [inline packages/gatsby-source-contentful/src/normalize.js]: add check for env variable to feature flag new feature

**Comment by Khaledgarbaya** [comment]
Hey @pieh I think this ready for review

**Comment by Simon-Tang** [comment]
Thanks @Khaledgarbaya for this! Is it currently possible to use this in GraphQL queries?I've looked at the [the link](https://www.contentful.com/developers/docs/concepts/rich-text/) to Contentful's docs and the changes in this PR but I haven't figured out what fields to use, if any are available right now.

That is, with a field called `richText`:

```graphql
... {
  richText {
     # Not sure what goes in here
  }
}
```

(as an aside, great unrelated [SO answer](https://stackoverflow.com/questions/51828234/graphql-queries-for-gatsby-contentful-setup-with-a-flexible-content-model)!)

**Comment by Khaledgarbaya** [comment]
Hi @Simon-Tang I am working on that right now https://github.com/contentful/gatsby-transformer-contentful-richtext/pull/1

**Pull request merged**